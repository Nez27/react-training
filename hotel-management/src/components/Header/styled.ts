import styled from 'styled-components';

const StyledHeader = styled.header`
  border-bottom: 1px solid var(--border-color);
  padding: 20px 40px;

  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 30px;
`;

export { StyledHeader };
