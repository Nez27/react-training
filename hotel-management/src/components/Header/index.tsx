// Components
import HeaderMenu from '../HeaderMenu';

// Styled
import { StyledHeader } from './styled';

interface IHeader {
  accountName: string
}

const Header = ({accountName}: IHeader) => {
  return (
    <StyledHeader>
      <p>Hi, {accountName}!</p>
      <HeaderMenu />
    </StyledHeader>
  );
};

export default Header;
