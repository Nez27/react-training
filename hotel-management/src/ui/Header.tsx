import styled from 'styled-components';
import HeaderMenu from './HeaderMenu';

const StyledHeader = styled.header`
  border-bottom: 1px solid var(--border-color);
  padding: 20px 40px;

  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 30px;
`;

const Header = () => {
  return (
    <StyledHeader>
      <p>Hi, Admin!</p>
      <HeaderMenu />
    </StyledHeader>
  );
};

export default Header;
