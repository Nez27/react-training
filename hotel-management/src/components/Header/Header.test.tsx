import renderer from 'react-test-renderer';

// Components
import Header from '.';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

describe('Header', () => {
  const queryClient = new QueryClient();

  const wrapper = renderer.create(
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Header accountName="Nezumi" />
      </BrowserRouter>
    </QueryClientProvider>
  );

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
