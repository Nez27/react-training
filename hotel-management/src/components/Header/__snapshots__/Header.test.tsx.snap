// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-dlniIP eqIruu"
      >
        <p>
          Hi, 
          Nezumi
          !
        </p>
        <ul
          class="sc-gtsqUy gOWXQs"
        >
          <li>
            <button
              aria-label="Profile"
              class="sc-bdnyFh bxHnoO"
              iconstyle="[object Object]"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M258.9 48C141.92 46.42 46.42 141.92 48 258.9c1.56 112.19 92.91 203.54 205.1 205.1 117 1.6 212.48-93.9 210.88-210.88C462.44 140.91 371.09 49.56 258.9 48zm126.42 327.25a4 4 0 01-6.14-.32 124.27 124.27 0 00-32.35-29.59C321.37 329 289.11 320 256 320s-65.37 9-90.83 25.34a124.24 124.24 0 00-32.35 29.58 4 4 0 01-6.14.32A175.32 175.32 0 0180 259c-1.63-97.31 78.22-178.76 175.57-179S432 158.81 432 256a175.32 175.32 0 01-46.68 119.25z"
                />
                <path
                  d="M256 144c-19.72 0-37.55 7.39-50.22 20.82s-19 32-17.57 51.93C191.11 256 221.52 288 256 288s64.83-32 67.79-71.24c1.48-19.74-4.8-38.14-17.68-51.82C293.39 151.44 275.59 144 256 144z"
                />
              </svg>
               
              <span />
            </button>
          </li>
          <li>
            <button
              aria-label="Logout"
              class="sc-bdnyFh bxHnoO"
              iconstyle="[object Object]"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
               
              <span />
            </button>
          </li>
        </ul>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-dlniIP eqIruu"
    >
      <p>
        Hi, 
        Nezumi
        !
      </p>
      <ul
        class="sc-gtsqUy gOWXQs"
      >
        <li>
          <button
            aria-label="Profile"
            class="sc-bdnyFh bxHnoO"
            iconstyle="[object Object]"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M258.9 48C141.92 46.42 46.42 141.92 48 258.9c1.56 112.19 92.91 203.54 205.1 205.1 117 1.6 212.48-93.9 210.88-210.88C462.44 140.91 371.09 49.56 258.9 48zm126.42 327.25a4 4 0 01-6.14-.32 124.27 124.27 0 00-32.35-29.59C321.37 329 289.11 320 256 320s-65.37 9-90.83 25.34a124.24 124.24 0 00-32.35 29.58 4 4 0 01-6.14.32A175.32 175.32 0 0180 259c-1.63-97.31 78.22-178.76 175.57-179S432 158.81 432 256a175.32 175.32 0 01-46.68 119.25z"
              />
              <path
                d="M256 144c-19.72 0-37.55 7.39-50.22 20.82s-19 32-17.57 51.93C191.11 256 221.52 288 256 288s64.83-32 67.79-71.24c1.48-19.74-4.8-38.14-17.68-51.82C293.39 151.44 275.59 144 256 144z"
              />
            </svg>
             
            <span />
          </button>
        </li>
        <li>
          <button
            aria-label="Logout"
            class="sc-bdnyFh bxHnoO"
            iconstyle="[object Object]"
          >
            <svg
              aria-hidden="true"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
             
            <span />
          </button>
        </li>
      </ul>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
